src/main/resources/static/data-mapping.js

```diff
@@ -1,11 +1,18 @@
-var uploaded_data;
-var uploaded_mapping;
-var usritems = [];
-var ocitems = [];
-var gridData = [];
+var usr_data;//store user-uploaded tabular data in JSON
+var oc_data;//store oc tree data in JSON
+var map_data;//store user-uploaded mapping data
+var usr_item_data;//store user-edited mapping data
+var selectedOCItem = null;//the selected OC item in the tree
+var selectedUsrItem = null;//the selected Usr item in the list
+var mapped_ocitems = [];//an array containing the paths of mapped oc items
+
 var baseSvg;
+var treeg, listg;
+
+var tipDiv;
+var leaf_depth = 5;
 var zoomListener;
-var hoveredItem = null;
+var rect_w = 100, rect_h = 15;//default rect size
```

Comment:
the rect widths are fixed, but they do scale with user's zoom actions. The text in a rect will be short-cut if it is too long, but will be shown in its full length upon user's mouse hover, in a tool tip